### ISSUE TITLE
Inconsistent isNew behaviour when calling beforeSave and afterSave

### ISSUE THREAD

**Issue by ben-eb** [HustleInc/parse-mockdb#67]
Hello,

I'm running into a problem when I register both `beforeSave` and `afterSave` hooks; my `afterSave` hook relies on the class to be saved. This isn't a problem when I test the hooks separately but I need to be able to run both hooks in other tests. I've got a simple reproduction that might help.

```js
const Parse = require('parse-shim');
const ParseMockDB = require('parse-mockdb');

class Target extends Parse.Object {
  constructor(attributes, options) {
    super('Target', attributes, options);
  }
}

Parse.Object.registerSubclass('Target', Target);

ParseMockDB.mockDB();

ParseMockDB.registerHook('Target', 'afterSave', (request) => {
  console.warn(`calling afterSave, isNew() => ${request.object.isNew()}`);
  return Parse.Promise.as(request.object);
});

ParseMockDB.registerHook('Target', 'beforeSave', (request) => {
  console.warn(`calling beforeSave, isNew() => ${request.object.isNew()}`);
  return Parse.Promise.as(request.object);
});

const target = new Target({
  title: 'Hello, world',
});

target.save()
  .then(() => {
    console.warn(`after hooks, isNew() => ${target.isNew()}`)
  })
  .then(() => {
    ParseMockDB.cleanUp();
    ParseMockDB.unMockDB();
  });
```

When I run this code, I get this output:

```
calling beforeSave, isNew() => true
calling afterSave, isNew() => true
after hooks, isNew() => false
```

However, when I remove the `beforeSave` handler, I get:

```
calling afterSave, isNew() => false
after hooks, isNew() => false
```

The expected behaviour is that *with* the `beforeSave` handler, the log should be:

```
calling beforeSave, isNew() => true
calling afterSave, isNew() => false
after hooks, isNew() => false
```

Let me know if you need any further details or if I can be of assistance.

**Comment by TylerBrock** [comment]
This should be resolved by v0.1.26 which we just released thanks to @EddyIonescu!

**Comment by ben-eb** [comment]
@TylerBrock @EddyIonescu Thanks guys! 🍻
